vnext/Desktop.ABITests/RedBoxTests.cpp

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include <winrt/Microsoft.Internal.h>
+#include <winrt/Microsoft.ReactNative.h>
+#include <winrt/Windows.Foundation.Collections.h>
+
+using namespace winrt;
+using namespace winrt::Microsoft::ReactNative;
+
+namespace ABITests {
+
+struct MockRedBoxHandler : implements<MockRedBoxHandler, Microsoft::ReactNative::IRedBoxHandler> {
+  void ShowNewError(IRedBoxErrorInfo info, RedBoxErrorType type) {
+    OnShowNewError(info, type);
+  }
+  bool IsDevSupportEnabled() {
+    return OnIsDevSupportEnabled();
+  };
+  void UpdateError(IRedBoxErrorInfo info) {
+    OnUpdateError(info);
+  }
+  void DismissRedBox() {
+    OnDismissRedBox();
+  }
+
+  std::function<void(IRedBoxErrorInfo, RedBoxErrorType)> OnShowNewError;
+  std::function<bool()> OnIsDevSupportEnabled;
+  std::function<void(IRedBoxErrorInfo)> OnUpdateError;
+  std::function<void()> OnDismissRedBox;
+};
+
+// The tests here are a development staging artifact owed to the incremental buildup of the C++/WinRT-based ABI of
+// the Win32 DLL. They (or their logical equivalent) should probably get rolled into other tests once C++/WinRT-based
+// instance management becomes available.
+TEST_CLASS (RedBoxTest) {
+ public:
+  TEST_METHOD(Handler_IsCallable) {
+    std::vector<std::tuple<IRedBoxErrorInfo, RedBoxErrorType>> showNewErrorCalls;
+    auto handler = make<MockRedBoxHandler>();
+    handler.as<MockRedBoxHandler>()->OnShowNewError = [&showNewErrorCalls](
+                                                          IRedBoxErrorInfo info, RedBoxErrorType type) {
+      showNewErrorCalls.emplace_back(info, type);
+    };
+    ReactInstanceSettings settings;
+    settings.RedBoxHandler(handler);
+
+    auto frameInfo1 = Microsoft::Internal::TestController::CreateRedBoxErrorFrameInfo(L"abc.js", L"foo", 123, 11);
+    auto frameInfo2 = Microsoft::Internal::TestController::CreateRedBoxErrorFrameInfo(L"def.js", L"bar", 234, 22);
+    auto errorInfo =
+        Microsoft::Internal::TestController::CreateRedBoxErrorInfo(L"out of coffee", 1203, {frameInfo1, frameInfo2});
+    settings.RedBoxHandler().ShowNewError(errorInfo, RedBoxErrorType::JavaScriptFatal);
```

Comment:
>RedBoxHandler [](start = 13, length = 13)

an I correct that this only tests the `RedBoxHandler` property of `ReactInstanceSettings`  since seemingly all this test does is call ShowNextError on the mock (albeit via the interface) and test the mock implementation...

Reply:
Correct. Not much test value, but I want to have at least some test that validates whether our .winmd contains what's needed to consume these interfaces.

Reply:
👍